rules/manager.go

```diff
@@ -589,8 +589,18 @@ func (g *Group) Eval(ctx context.Context, ts time.Time) {
 			)
 
 			app := g.opts.Appendable.Appender()
+			var appErr error
 			seriesReturned := make(map[string]labels.Labels, len(g.seriesInPreviousEval[i]))
 			defer func() {
+				if appErr != nil {
+					app.Rollback()
+					app = g.opts.Appendable.Appender()
```

Comment:
@codesome hence we take a new appender

Reply:
Ah right, didn't notice this reassignment